plugins/jobtype/test/azkaban/jobtype/TestHadoopTuningConfigurationInjector.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.jobtype;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.MRJobConfig;
+import org.junit.Assert;
+import org.junit.Test;
+
+import azkaban.flow.CommonJobProperties;
+import azkaban.utils.Props;
+
+import com.google.common.io.Files;
+
+
+public class TestHadoopTuningConfigurationInjector {
+
+  @Test
+  public void testPrepareResourcesToInject() throws IOException {
+
+    File root = Files.createTempDir();
+
+    Props allProps = new Props();
+    Props azkabanProps = new Props();
+    Props userProps = new Props();
+
+    userProps.put(HadoopConfigurationInjector.INJECT_PREFIX + MRJobConfig.IO_SORT_MB, "100");
+    userProps.put(HadoopConfigurationInjector.INJECT_PREFIX + MRJobConfig.IO_SORT_FACTOR, "100");
+    userProps.put(HadoopConfigurationInjector.INJECT_PREFIX + MRJobConfig.REDUCE_MEMORY_MB, "2048");
+    userProps.put(HadoopConfigurationInjector.INJECT_PREFIX + MRJobConfig.MAP_MEMORY_MB, "2048");
+
+    azkabanProps.put(CommonJobProperties.EXEC_ID, CommonJobProperties.EXEC_ID);
```

Comment:
I think I mentioned this in the last review, but don't see my comment / a response - can we set all of these properties via a loop?

Reply:
Done.